src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -866,8 +886,52 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression? receiverOpt,
                 return false;
             }
 
+            if (delegateType.IsFunctionPointer() && !method.IsStatic)
+            {
+                Error(diagnostics, ErrorCode.ERR_FuncPtrMethMustBeStatic, errorLocation, method);
```

Comment:
>Error(diagnostics, ErrorCode.ERR_FuncPtrMethMustBeStatic, errorLocation, method); [](start = 16, length = 81)

It feels like we are duplicating error reporting with OverloadResolutionResult.cs. Why do we have to do this? Are both code paths covered by tests?

Reply:
> It feels like we are duplicating error reporting with OverloadResolutionResult.cs.

Unfortunately, we are. This is the purpose of the comment above: this method calculates function pointer compatibility, instead of overload applicability. Theoretically, this code path cannot be hit. It's here for the purpose of completely implementing the spec as written, but I could replace it with an assert if you prefer.

Reply:
> but I could replace it with an assert

Let's add (rather than replace) an assert with a comment to add a test if the code path becomes reachable.

---
In reply to: [397477684](https://github.com/dotnet/roslyn/pull/42488#discussion_r397477684) [](ancestors = 397477684)